### PR TITLE
Add a slack link in the application footer

### PR DESCRIPTION
**Pull request by WilliamZapata**
Update header and footer to accept attributes.
Update all pages' test snapshots (all use the footer).

### CODE REVIEW

File: alluxio-ui/master/src/containers/App/__snapshots__/App.test.tsx.snap [hunk 1]
```diff
@@ -1153,6 +1153,13 @@ exports[`App Connected component Matches snapshot 1`] = `
                           "innerText": "Resources",
                           "url": "https://alluxio.org/alluxio-resources/",
                         },
+                        Object {
+                          "attributes": Object {
+                            "target": "_blank",
+                          },
+                          "innerText": "Slack Channel",
+                          "url": "https://slackin.alluxio.org/",
```
Comment (apc999) [inline alluxio-ui/master/src/containers/App/__snapshots__/App.test.tsx.snap]: can we use "https://www.alluxio.org/slack"? slackin is the implementation details to distribute slack invitation code

File: alluxio-ui/master/src/containers/App/__snapshots__/App.test.tsx.snap [hunk 2]
```diff
@@ -1272,6 +1279,29 @@ exports[`App Connected component Matches snapshot 1`] = `
...
+                              className="nav-item"
+                            >
+                              <NavLink
+                                href="https://slackin.alluxio.org/"
+                                tag="a"
+                                target="_blank"
+                              >
+                                <a
+                                  className="nav-link"
+                                  href="https://slackin.alluxio.org/"
```
Comment (apc999) [inline alluxio-ui/master/src/containers/App/__snapshots__/App.test.tsx.snap]: ditto

File: alluxio-ui/master/src/containers/App/__snapshots__/App.test.tsx.snap [hunk 3]
```diff
@@ -1447,6 +1477,13 @@ exports[`App Shallow component Matches snapshot 1`] = `
               "innerText": "Resources",
               "url": "https://alluxio.org/alluxio-resources/",
             },
+            Object {
+              "attributes": Object {
+                "target": "_blank",
+              },
+              "innerText": "Slack Channel",
+              "url": "https://slackin.alluxio.org/",
```
Comment (apc999) [inline alluxio-ui/master/src/containers/App/__snapshots__/App.test.tsx.snap]: ditto

File: alluxio-ui/master/src/containers/App/__snapshots__/App.test.tsx.snap (hunk 1 above)
Comment (ZacBlanco) [reply alluxio-ui/master/src/containers/App/__snapshots__/App.test.tsx.snap]: https://www.alluxio.org/slack redirects to https://slackin.alluxio.org. I don't think it makes a difference

File: alluxio-ui/master/src/constants/data/footerNavigation.tsx [hunk 4]
```diff
@@ -26,4 +26,8 @@ export const footerNavigationData : INavigationData[] = [{
 }, {
   innerText: 'Resources',
   url: 'https://alluxio.org/alluxio-resources/'
+}, {
+  attributes: {target: '_blank'},
```
Comment (calvinjia) [inline alluxio-ui/master/src/constants/data/footerNavigation.tsx]: Slightly unrelated to this PR, but should the other links also open in a new tab?

File: alluxio-ui/master/src/containers/App/__snapshots__/App.test.tsx.snap (hunk 1 above)
Comment (WilliamZapata) [reply alluxio-ui/master/src/containers/App/__snapshots__/App.test.tsx.snap]: I'm ok with either way. I'll make @apc999's suggestion happen.

File: alluxio-ui/master/src/constants/data/footerNavigation.tsx (hunk 4 above)
Comment (WilliamZapata) [reply alluxio-ui/master/src/constants/data/footerNavigation.tsx]: Oh, good catch, I'll take a look and open them in a new tab if they make sense.

**Comment by WilliamZapata** [comment]
Jenkins, test this please.

**Pull request merged**